types/mithril/test/test-class-component.ts

```diff
@@ -5,6 +5,7 @@ import * as m from 'mithril';
 // Simplest component example - no attrs or state.
 //
 class Comp0 implements m.ClassComponent {
+    [_: number]: any;
```

Comment:
What's the point of these?

Reply:
4.3 requires an index signature when the constraint, Lifecycle in this case, has an index signature.

Reply:
Without the index signature, Typescript (correctly) complains that types constrained to Lifecycle have no properties in common with it. The compiler requires SOME evidence of common properties.

For example, in test-api.ts: `m.Comp<{}, { data: string }>` has `{ data: string }` constrained to `Lifecycle<{}, { data: string }>`, but that implies that it has at least one of the on* methods, which it doesn't. The index signature works around this, but is not a great workaround in my opinion.

Reply:
Oh, in that case, try removing the property from `Lifecycle` and see if that fixes the issue. If that fails, then I'll need to figure out something on my own. I'm short on cycles ATM, so keep that in mind.